File: server/router/business/cmsDetail.go
package business

import (
	"go-cms/api/v1"
	"go-cms/middleware"
	"github.com/gin-gonic/gin"
)

type CmsDetailRouter struct {
}

// InitCmsDetailRouter 初始化 CmsDetail 路由信息
func (s *CmsDetailRouter) InitCmsDetailRouter(Router *gin.RouterGroup) {
	router := Router.Group("cmsDetail").Use(middleware.OperationRecord())
	routerNoRecord := Router.Group("cmsDetail")
	var apiV1 = v1.ApiGroupApp.BusinessApiGroup.CmsDetailApi
	{
		router.POST("createCmsDetail", apiV1.CreateCmsDetail)   // 新建CmsDetail
		router.POST("deleteCmsDetail", apiV1.DeleteCmsDetail) // 删除CmsDetail
		router.POST("deleteCmsDetailByIds", apiV1.DeleteCmsDetailByIds) // 批量删除CmsDetail
		router.POST("updateCmsDetail", apiV1.UpdateCmsDetail)    // 更新CmsDetail
	    router.POST("quickEdit", apiV1.QuickEdit)  // 快速编辑	
	}
	{
		routerNoRecord.GET("findCmsDetail", apiV1.FindCmsDetail)        // 根据ID获取CmsDetail
		routerNoRecord.GET("getCmsDetailList", apiV1.GetCmsDetailList)  // 获取CmsDetail列表
        routerNoRecord.GET("excelList", apiV1.ExcelList)  // 分页导出excel CmsDetail列表
	}
}

File: web/src/api/cmsDetail.ts
import { http,resData } from "@/utils/http";
 
// @Tags CmsDetail
// @Summary 创建CmsDetail
export const createCmsDetail = (data?: object) => {
   return http.request<resData>("post", "/cmsDetail/createCmsDetail", { data });  
}

// @Tags CmsDetail
// @Summary 删除CmsDetail
export const deleteCmsDetail = (data?: object) => {
   return http.request<resData>("post", "/cmsDetail/deleteCmsDetail", { data });  
}
 

// @Tags CmsDetail
// @Summary 删除ByIdsCmsDetail
export const deleteCmsDetailByIds = (data?: object) => {
   return http.request<resData>("post", "/cmsDetail/deleteCmsDetailByIds", { data });  
}
 

// @Tags CmsDetail
// @Summary 更新CmsDetail
export const updateCmsDetail = (data?: object) => {
   return http.request<resData>("post", "/cmsDetail/updateCmsDetail", { data });  
}

// @Tags CmsDetail
// @Summary 快速编辑
export const quickEdit = (data?: object) => {
   return http.request<resData>("post", "/cmsDetail/quickEdit", { data });  
}
 

// @Tags CmsDetail
// @Summary 用id查询CmsDetail
export const findCmsDetail = (params?: object) => {
   return http.request<resData>("get", "/cmsDetail/findCmsDetail", { params });  
}
 

// @Tags CmsDetail
// @Summary 分页获取CmsDetail列表
export const getCmsDetailList = (params?: object) => {
   return http.request<resData>("get", "/cmsDetail/getCmsDetailList", { params });  
}
 

// @Tags CmsDetail
// @Summary 导出excelCmsDetail列表
export const excelList = (params?: object) => {
   return http.request<resData>("get", "/cmsDetail/excelList", { params });  
}
